The job **`pr-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26649222613/job/78543035261) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/78543035261) [(plain)](https://github.com/rust-lang/rust/commit/a6d5705eca9766ee5f65d6436f1505c750c28a9b/checks/78543035261/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking toml v0.7.8
error[E0004]: non-exhaustive patterns: `&rustc_hir::PatKind::CVarArgs` not covered
    --> src/tools/clippy/clippy_utils/src/hir_utils.rs:1462:15
     |
1462 |         match &pat.kind {
     |               ^^^^^^^^^ pattern `&rustc_hir::PatKind::CVarArgs` not covered
     |
note: `rustc_hir::PatKind<'_>` defined here
    --> compiler/rustc_hir/src/hir.rs:2104:1
     |
2104 | pub enum PatKind<'hir> {
     | ^^^^^^^^^^^^^^^^^^^^^^
...
2174 |     CVarArgs,
     |     -------- not covered
     = note: the matched value is of type `&rustc_hir::PatKind<'_>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
     |
1532 ~             PatKind::Never | PatKind::Wild | PatKind::Err(_) => {},
1533 ~             &rustc_hir::PatKind::CVarArgs => todo!(),
     |

[RUSTC-TIMING] toml test:false 0.354
[RUSTC-TIMING] winnow test:false 1.280
    Checking serde_spanned v1.1.0
---
     |     -------- not covered
     = note: the matched value is of type `rustc_hir::PatKind<'_>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
     |
1571 ~         PatKind::Expr(..) | PatKind::Range(..) | PatKind::Err(_) | PatKind::Deref(_) | PatKind::Guard(..) => true,
1572 ~         rustc_hir::PatKind::CVarArgs => todo!(),
     |

[RUSTC-TIMING] icu_properties test:false 1.104
    Checking idna_adapter v1.2.1
[RUSTC-TIMING] idna_adapter test:false 0.059
```

</details>

--- a/src/librustdoc/html/macro_expansion.rs
+++ b/src/librustdoc/html/macro_expansion.rs
@@ -2,7 +2,7 @@ use rustc_ast::visit::{
     AssocCtxt, Visitor, walk_assoc_item, walk_crate, walk_expr, walk_item, walk_pat, walk_stmt,
     walk_ty,
 };
-use rustc_ast::{AssocItem, Crate, Expr, Item, Pat, Stmt, Ty};
+use rustc_ast::{AssocItem, Crate, Expr, ForeignItem, Item, Pat, Stmt, Ty};
 use rustc_data_structures::fx::FxHashMap;
 use rustc_span::source_map::SourceMap;
 use rustc_span::{BytePos, Span};
@@ -172,6 +172,16 @@ impl<'ast> Visitor<'ast> for ExpandedCodeVisitor<'ast> {
             });
         } else {
             walk_assoc_item(self, item, ctxt);
+        }
+    }
+
+    fn visit_foreign_item(&mut self, item: &'ast ForeignItem) -> Self::Result {
+        if item.span.from_expansion() {
+            self.handle_new_span(item.span, || {
+                rustc_ast_pretty::pprust::foreign_item_to_string(item)
+            });
+        } else {
+            walk_item(self, item);
         }
     }
 }

--- a/tests/rustdoc-html/macro-expansion/c-var-args.rs
+++ b/tests/rustdoc-html/macro-expansion/c-var-args.rs
@@ -1,0 +1,19 @@
+// Ensure that C var args (`va_list`) work.
+// Regression test for <https://github.com/rust-lang/rust/issues/156486>.
+
+//@ compile-flags: -Zunstable-options --generate-macro-expansion
+
+#![crate_name = "foo"]
+
+//@ has 'src/foo/c-var-args.rs.html'
+
+macro_rules! print {
+    () => {
+        fn printf(...);
+    };
+}
+
+//@ has - '//*[@class="expansion"]/*[@class="expanded"]' 'fn printf(...);'
+extern "C" {
+    print! {}
+}

